docs/spec/consensus/light-client.md

```diff
@@ -1,113 +1,316 @@
-# Light Client
-
-A light client is a process that connects to the Tendermint Full Node(s) and then tries to verify the Merkle proofs
-about the blockchain application. In this document we describe mechanisms that ensures that the Tendermint light client
-has the same level of security as Full Node processes (without being itself a Full Node).
-
-To be able to validate a Merkle proof, a light client needs to validate the blockchain header that contains the root app hash.
-Validating a blockchain header in Tendermint consists in verifying that the header is committed (signed) by >2/3 of the
-voting power of the corresponding validator set. As the validator set is a dynamic set (it is changing), one of the
-core functionality of the light client is updating the current validator set, that is then used to verify the
-blockchain header, and further the corresponding Merkle proofs.
-
-For the purpose of this light client specification, we assume that the Tendermint Full Node exposes the following functions over
-Tendermint RPC:
-
-```golang
-Header(height int64) (SignedHeader, error) // returns signed header for the given height
-Validators(height int64) (ResultValidators, error) // returns validator set for the given height
-LastHeader(valSetNumber int64) (SignedHeader, error)   // returns last header signed by the validator set with the given validator set number
-
-type SignedHeader struct {
-    Header        Header
-    Commit        Commit
-    ValSetNumber  int64
-}
+# Lite client
 
-type ResultValidators struct {
-    BlockHeight   int64
-    Validators    []Validator
-    // time the current validator set is initialised, i.e, time of the last validator change before header BlockHeight
-    ValSetTime    int64
-}
-```
+A lite client is a process that connects to Tendermint full nodes and then tries to verify application data using the Merkle proofs.
+
+## Context of this document
+
+In order to make sure that full nodes have the incentive to follow the protocol, we have to address the following three Issues
+
+1) The lite clients needs a method to verify header it obtains from full nodes -- addressed in this document)
+2) Detect conflicts/forged headers -- when the verification method of this document fails, then a conflict is detected. However, this document does not address to do this in an efficient way, or with special attacks in mind. This will be treated in fork accountability.
```

Comment:
2) The lite client must be able to connect to one correct full node to detect and report on failures in the trust assumptions (ie. conflicting headers) -- a future document.

Reply:
I incorporated the suggested text. I am not sure whether the addition was intended that way. Please check!